lib/promscrape/scrapework.go

```diff
@@ -1050,12 +1080,17 @@ func (wc *writeRequestCtx) addRow(cfg *ScrapeWork, r *parser.Row, timestamp int6
 	wc.labels = promrelabel.FinalizeLabels(wc.labels[:labelsLen], wc.labels[labelsLen:])
 	if len(wc.labels) == labelsLen {
 		// Skip row without labels.
-		return
+		return nil
 	}
 	// Add labels from `global->external_labels` section after the relabeling like Prometheus does.
 	// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3137
 	externalLabels := cfg.ExternalLabels.GetLabels()
 	wc.labels = appendExtraLabels(wc.labels, externalLabels, labelsLen, cfg.HonorLabels)
+	labelLimit := cfg.LabelLimit
+	if needRelabel && labelLimit > 0 && len(wc.labels)-labelsLen > labelLimit {
```

Comment:
It's unclear why labelLimit check should be applied only if `needRelabel` is set.  I suggest to remove this implicit dependency.